### DISCUSSION TITLE
Referencing the source filename -- What am I doing wrong?

### DISCUSSION THREAD

**Discussion by gaiking-uk** [ImageMagick/ImageMagick#7836] [Help] [answered]
### ImageMagick version

7.1.1-41

### Operating system, version and so on

Windows 11

### Context
- I have a folder of icons: `D:\Icons\Grid`
- Files are named: `Apple.png`, `Box.png`, `Cog.png`, etc
- I want to take every `.png` in `D:\Icons\Grid`, apply a simple transform (in this case add padding) and then save the `.png` in a parallel folder, `D:\Icons\Grid2`

![Magick exe -- files won't name correctly](https://github.com/user-attachments/assets/009d4c10-1315-4b6a-8f47-781afda88c0c)

### Issue
- I ran variations of the following command...
`magick.exe "D:\Icons\Grid\*.png" -format png -bordercolor #288647 -border 108x47 "D:\Icons\Grid2\*.png"`
- However instead of saving the files in `D:\Icons\Grid2` as `Apple.png`, `Box.png`, `Cog.png`, Magick.exe kept saving them as: `%t-0.png`, `%t-1.png`, `%t-2.png`, etc

### Solutions tried
- I tried only specifying the output folder (`"D:\Icons\Grid2\"` instead of `"D:\Icons\Grid2\*.png"`) but Magick.exe didn't seem to like that
- I tried escaping the `%t` parameter by passing `%%t`, `^%t` instead, but these characters were just passed into the output filenames as literals
- I tried using other short parameters instead (`%e`, `%d`, etc) but this didn't work
- I tried using the long parameter format (`%[basename]`, etc) but once again, this was simply converted to the output filename as a literal string

___

### What am I doing wrong!?

**I know I will kick myself as will no doubt turn out to be something stupid and obvious, but can someone please tell me what I'm doing wrong! 🤦🏼‍♂️**

**Comment by GeeMack** [comment, marked as answer]
ImageMagick has a special way of handling a variable that holds a filename. As this command reads all the input files, it reads the name of each one into a variable that rides along with the image and can be used as the output name. Notice the use of "-set filename:0 "%[t]" in this command...

    magick.exe "D:\Icons\Grid\*.png" -set filename:0 "%[t]" ^
       -format png -bordercolor #288647 -border 108x47 "D:\Icons\Grid2\%[filename:0].png"

**Comment by snibgo** [comment]
As GeeMack says.

In addition, I suggest you remove "-format png". That is correct only for "magick mogrify". For just plain "magick", the "-format" option is for writing to "info:", which you are not doing. So your "-format" has no effect and is harmless but misleading.

**Comment by gaiking-uk** [comment]
Thanks @GeeMack! -- Was probably banging my head against a wall for the better part of an hour on this, but happy to report your code worked perfectly! 👍🏼

Also @snibgo -- Thanks for your suggestion, I included the `-format png` based on an [example script I found online](https://legacy.imagemagick.org/discourse-server/viewtopic.php?t=36839) but as you suggested, this used `mogrify` so appreciate the info and advice to take this out and have done this too, thanks! 👍🏼

**Reply by fmw42** [reply]
That command in the linked page was for mogrify, not convert or magick.
